### PR TITLE
[RISCV][clang] Add Zvfbfwma C intrinsics support

### PR DESCRIPTION
**Pull request by 4vtomat**

### CODE REVIEW

File: clang/include/clang/Basic/riscv_vector.td [hunk 1]
```diff
@@ -1730,12 +1730,26 @@ let ManualCodegen = [{
     defm vfwnmacc : RVVFloatingWidenTerBuiltinSetRoundingMode;
     defm vfwmsac  : RVVFloatingWidenTerBuiltinSetRoundingMode;
     defm vfwnmsac : RVVFloatingWidenTerBuiltinSetRoundingMode;
+
+    // Vector BF16 widening multiply-accumulate
+    let Log2LMUL = [-2, -1, 0, 1, 2],
```
Comment (topperc) [inline clang/include/clang/Basic/riscv_vector.td:1735]: Do we need a RequiredFeatures for this?

File: clang/test/CodeGen/RISCV/rvv-intrinsics-autogenerated/non-policy/non-overloaded/vfwmaccbf16.c [hunk 2]
```diff
@@ -0,0 +1,479 @@
+// NOTE: Assertions have been autogenerated by utils/update_cc_test_checks.py UTC_ARGS: --version 4
+// REQUIRES: riscv-registered-target
+// RUN: %clang_cc1 -triple riscv64 -target-feature +v -target-feature +zfh \
+// RUN:   -target-feature +zvfh -target-feature +experimental-zvfbfwma -disable-O0-optnone  \
```
Comment (tclin914) [inline clang/test/CodeGen/RISCV/rvv-intrinsics-autogenerated/non-policy/non-overloaded/vfwmaccbf16.c:4]: Are `-target-feature +zfh` and `-target-feature +zvfh` necessary?

Comment (4vtomat) [reply clang/test/CodeGen/RISCV/rvv-intrinsics-autogenerated/non-policy/non-overloaded/vfwmaccbf16.c:4]: No, it doesn't, I just copied the test cases from [this](https://github.com/riscv-non-isa/rvv-intrinsic-doc/pull/293) lol. I will remove it though.

File: clang/include/clang/Basic/riscv_vector.td (hunk 1 above)
Comment (4vtomat) [reply clang/include/clang/Basic/riscv_vector.td:1735]: Yes, we do.

**Review by tclin914 — APPROVED** [review]
LGTM

**Review by topperc — APPROVED** [review]
LGTM

**Pull request merged**